src/portals/Screencopy.cpp

```diff
@@ -271,6 +271,12 @@ static void hlOnBufferDone(void* data, hyprland_toplevel_export_frame_v1* frame)
         PSESSION->sharingData.windowFrameCallback = nullptr;
         Debug::log(LOG, "[screencopy/pipewire] Out of buffers");
         PSESSION->sharingData.status = FRAME_NONE;
+        if (PSESSION->sharingData.copyRetries++ < MAX_RETRIES) {
+            Debug::log(LOG, "[sc] Retrying screencopy ({}/{})", PSESSION->sharingData.copyRetries, MAX_RETRIES);
+            g_pPortalManager->m_sPortals.screencopy->m_pPipewire->updateStreamParam(PSTREAM);
+            g_pPortalManager->m_sPortals.screencopy->queueNextShareFrame(PSESSION);
+        }
+        PSESSION->sharingData.copyRetries = 0;
```

Comment:
wait no this is wrong, it should be a few lines below, right before hyprland_toplevel_export_frame_v1_copy

Reply:
My bad. I moved the line 279 where you pointed.